src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -244,6 +244,37 @@ protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string n
             return symbolsMatch;
         }
 
+        internal static bool OriginalSymbolsMatch(
+            ISymbol searchSymbol,
+            ISymbol symbolToMatch,
+            Solution solution,
+            CancellationToken cancellationToken)
+        {
+            if (ReferenceEquals(searchSymbol, symbolToMatch))
+            {
+                return true;
+            }
+
+            if (searchSymbol == null || symbolToMatch == null)
+            {
+                return false;
+            }
+
+            Compilation searchSymbolCompilation = null;
+            if (!TryGetCompilation(searchSymbol, solution, ref searchSymbolCompilation, cancellationToken))
+            {
+                return false;
+            }
+
+            Compilation symbolToMatchCompilation = null;
+            if (!TryGetCompilation(symbolToMatch, solution, ref symbolToMatchCompilation, cancellationToken))
+            {
+                return false;
+            }
+
+            return OriginalSymbolsMatch(searchSymbol, symbolToMatch, solution, ref searchSymbolCompilation, symbolToMatchCompilation, cancellationToken);
```

Comment:
Short answer: I don't know.

Longer possible answer: eventually `OriginalSymbolsMatch()` calls `TryGetCompilation()` on `searchSymbolCompilation` which requires a `ref` (see earlier comment).  As to why only `searchSymbolCompilation` is `ref` and `symbolToMatchCompilation` isn't, ¯\\(°_o)/¯.